jupyter-tensorflow-cuda-full/rockcraft.yaml

```diff
@@ -0,0 +1,244 @@
+      echo "deb https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64 /" > /etc/apt/sources.list.d/nvidia-ml.list
+      echo "deb https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64 /" > /etc/apt/sources.list.d/nvidia-ml.list
+      apt-get -yq update
+      apt-get -yq install --no-install-recommends cm-super cuda-command-line-tools-${CUDA_VERSION/./-} cuda-nvrtc-${CUDA_VERSION/./-} libcublas-${CUDA_VERSION/./-} libcudnn8=${CUDNN_VERSION}+cuda${CUDA_VERSION} libcufft-${CUDA_VERSION/./-} libcurand-${CUDA_VERSION/./-} libcusolver-${CUDA_VERSION/./-} libcusparse-${CUDA_VERSION/./-} libfreetype6-dev libhdf5-serial-dev libnvinfer7=${LIBNVINFER_VERSION}+cuda${OLD_CUDA_VERSION} libnvinfer-plugin7=${LIBNVINFER_VERSION}+cuda${OLD_CUDA_VERSION} libzmq3-dev pkg-config
+      apt-get clean
+      rm -rf /var/lib/apt/lists/*
+
+      # copy other nvidia stuff directories
+      cp -rp /usr/local/cuda-${CUDA_VERSION}/* ${CRAFT_PART_INSTALL}/usr/local/cuda-${CUDA_VERSION}
+
+      # tensorflow fix - wrong libcuda lib path (+ reconfigure dynamic linker run-time bindings)
+      ln -s /usr/local/cuda/lib64/stubs/libcuda.so ${CRAFT_PART_INSTALL}/usr/local/cuda/lib64/stubs/libcuda.so.1
+      echo "/usr/local/cuda/lib64/stubs" > /etc/ld.so.conf.d/z-cuda-stubs.conf
+      ldconfig
+
+      # copy CUDA config
+      mkdir -p ${CRAFT_PART_INSTALL}/etc/ld.so.conf.d/
+      cp -r /etc/ld.so.conf.d/z-cuda-stubs.conf ${CRAFT_PART_INSTALL}/etc/ld.so.conf.d/z-cuda-stubs.conf
+
+      # tensorflow fix - wrong libcusolver lib path
+      # https://github.com/tensorflow/tensorflow/issues/43947#issuecomment-748273679
+      ln -s /usr/local/cuda-${CUDA_VERSION}/targets/x86_64-linux/lib/libcusolver.so.11 ${CRAFT_PART_INSTALL}/usr/local/cuda-${CUDA_VERSION}/targets/x86_64-linux/lib/libcusolver.so.10
+
+      # tensorflow fix - some tensorflow tools expect a `python` binary
+      ln -s /usr/local/bin/python3 ${CRAFT_PART_INSTALL}/usr/local/bin/python
+
+      # Install the jupyter-tensorflow CUDA requirements
+      cp ${CRAFT_PART_SRC}/components/example-notebook-servers/jupyter-tensorflow/cuda-requirements.txt cuda-requirements.txt
+      "${CONDA_BIN}/pip" install --no-cache-dir -r cuda-requirements.txt
+
+      # Install the jupyter-tensorflow-full requirements
+      cp ${CRAFT_PART_SRC}/components/example-notebook-servers/jupyter-tensorflow-full/requirements.txt requirements.txt
+      "${CONDA_BIN}/pip" install --no-cache-dir -r requirements.txt
+
+      # Write some config files for activating conda in user sessions
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${HOME}/.bashrc"
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${CRAFT_PART_INSTALL}/etc/profile"
+      echo "conda activate base" >> "${HOME}/.bashrc"
+      echo "conda activate base" >> "${CRAFT_PART_INSTALL}/etc/profile"
+
+      # Do some cleanup
+      rm -rf "${HOME}/.wget-hsts" "${HOME}/.cache"
```

Comment:
it's hosts no ?

Reply:
Yes, it is in build. Again this came verbatim from the Dockerfile.